### PR TITLE
[Prevoker] Bare bones implementation of Evoker guide

### PR DESCRIPTION
**Pull request by trevorm4**
### Description

Adds DB,SB,DF, TA sections to guide

### Testing

- Test report URL:  report/FTV9GmMw6bD3Xvtc/32-Mythic+Terros+-+Kill+(5:20)/479-Trevordwagon/standard
- Screenshot(s): image

### CODE REVIEW

File: src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx [hunk 1]
```diff
@@ -1,39 +1,83 @@
...
+  onCast(event: EmpowerEndEvent) {
+    this.totalBreaths += 1;
+    const info = {
+      timestamp: event.timestamp,
+      coyActive: this.selectedCombatant.hasBuff(SPELLS.CALL_OF_YSERA_BUFF.id),
+      targetsHit: 0,
+      counted: false,
+    };
+    // ignore cancelled casts
+    if (this.casts.at(-1)?.targetsHit === 0) {
```
Comment (vohrr) [inline src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx]: EmpowerEnd is an extension of BaseCast so it should have an isCancelled property you could check

File: src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx [hunk 2]
```diff
@@ -48,6 +92,81 @@ class DreamBreath extends Analyzer {
...
+
+    const entries: BoxRowEntry[] = [];
+    this.casts.forEach((cast) => {
+      let value = QualitativePerformance.Fail;
+      const coyActive =
+        this.selectedCombatant.hasTalent(TALENTS_EVOKER.CALL_OF_YSERA_TALENT) && cast.coyActive;
+      if (coyActive && cast.targetsHit === 5) {
+        value = QualitativePerformance.Good;
+      } else if (coyActive && cast.targetsHit === 4) {
+        value = QualitativePerformance.Ok;
```
Comment (vohrr) [inline src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx:114]: Is it always considered a bad cast if they use DB without the CoY buff, even if they hit 5 targets?

File: src/analysis/retail/evoker/preservation/modules/talents/DreamFlight.tsx [hunk 3]
```diff
@@ -60,6 +90,93 @@ class DreamFlight extends Analyzer {
...
+            const percentHit = info.totalHit / this.combatants.playerCount;
+            if (percentHit < 0.65) {
+              targetsHitPerf = QualitativePerformance.Fail;
+            } else if (percentHit < 0.75) {
+              targetsHitPerf = QualitativePerformance.Ok;
+            }
+            checklistItems.push({
+              label: (
+                <>
+                  % of raid hit with <SpellLink spell={TALENTS_EVOKER.DREAM_FLIGHT_TALENT} />
```
Comment (vohrr) [inline src/analysis/retail/evoker/preservation/modules/talents/DreamFlight.tsx]: Might be more readable to show it as targetsHit / playerCount i.e. `16/20 Targets` or  `16 out of 20 Targets` etc

File: src/analysis/retail/evoker/preservation/modules/talents/DreamFlight.tsx [hunk 4]
```diff
@@ -60,6 +90,93 @@ class DreamFlight extends Analyzer {
...
+                </>
+              ),
+              result: <PerformanceMark perf={targetsHitPerf} />,
+              details: <>{formatPercentage(percentHit)}%</>,
+            });
+            const overhealPercent = info.overhealing / (info.overhealing + info.healing);
+            if (overhealPercent > 0.6) {
+              overhealingPerf = QualitativePerformance.Fail;
+            } else if (overhealPercent > 0.5) {
```
Comment (vohrr) [inline src/analysis/retail/evoker/preservation/modules/talents/DreamFlight.tsx]: 50% overheal seems high for OK on a raid CD, esp. after the health change but I'll defer to your judgment

File: src/analysis/retail/evoker/preservation/modules/talents/Spiritbloom.tsx [hunk 5]
```diff
@@ -29,6 +35,48 @@ class DreamBreath extends Analyzer {
...
+        is one of your empowered abilities and a very strong AoE triage heal that applies a powerful
+        HoT with the Abberus (T30) tier set. You should try to use this ability at maximum
+        Empowerment level whenever it is not on cooldown.{' '}
+        <SpellLink id={TALENTS_EVOKER.SPIRITBLOOM_TALENT} /> is a strong candidate to consume{' '}
+        <SpellLink spell={TALENTS_EVOKER.ECHO_TALENT} /> with, especially with the T30 tierset, but
+        may vary based on playstyle.
+      </p>
+    );
+
+    const data = (
```
Comment (vohrr) [inline src/analysis/retail/evoker/preservation/modules/talents/Spiritbloom.tsx:77]: I think a qualitative performance bar similar to the dream breath guideSubsection would be good here, given the importance of hitting full empowered SBs bc of the tier set. Won't hold the pr up for it though

**Review by vohrr — APPROVED** [review]
Some comments but overall looks good

File: src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx (hunk 2 above)
Comment (trevorm4) [reply src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx:114]: Yes, not using Call of Ysera is inting.

File: src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx (hunk 1 above)
Comment (trevorm4) [reply src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx]: It extends basecastevent so it doesn't have that property, but turned out I didn't need this check after switching from CastEvent to EmpowerEnd.

**Pull request merged**